apps/web/src/pages/WorkStreams/CreateWorkStreamPage.tsx

```diff
@@ -0,0 +1,309 @@
+import { useNavigate } from "react-router";
+import { SubmitHandler } from "react-hook-form";
+import { useIntl } from "react-intl";
+import { useMutation, useQuery } from "urql";
+import IdentificationIcon from "@heroicons/react/24/outline/IdentificationIcon";
+
+import { toast } from "@gc-digital-talent/toast";
+import {
+  BasicForm,
+  Input,
+  OptGroupOrOption,
+  Select,
+  Submit,
+} from "@gc-digital-talent/forms";
+import { commonMessages, errorMessages } from "@gc-digital-talent/i18n";
+import {
+  graphql,
+  Scalars,
+  CreateWorkStreamInput,
+  LocalizedStringInput,
+  InputMaybe,
+} from "@gc-digital-talent/graphql";
+import { ROLE_NAME } from "@gc-digital-talent/auth";
+import {
+  Heading,
+  Link,
+  CardSeparator,
+  CardBasic,
+  Pending,
+} from "@gc-digital-talent/ui";
+import { unpackMaybes } from "@gc-digital-talent/helpers";
+
+import SEO from "~/components/SEO/SEO";
+import useRoutes from "~/hooks/useRoutes";
+import useBreadcrumbs from "~/hooks/useBreadcrumbs";
+import RequireAuth from "~/components/RequireAuth/RequireAuth";
+import pageTitles from "~/messages/pageTitles";
+import Hero from "~/components/Hero";
+import adminMessages from "~/messages/adminMessages";
+
+const CreateWorkStream_Mutation = graphql(/* GraphQL */ `
+  mutation CreateWorkStream($workStream: CreateWorkStreamInput!) {
+    createWorkStream(workStream: $workStream) {
+      id
+    }
+  }
+`);
+
+interface FormValues {
+  key?: InputMaybe<Scalars["String"]["input"]>;
```

Comment:
I see this in the form values but no way of changing it and it does not get submitted. 

@tristan-orourke How is this meant to work? Is it going to be:

1.  a required field like on the skill family,
2. an optional field
3. Not included at all and auto generated from name
4. auto generated from name but show as a field that can be changed

Reply:
Hmm. I no longer feel so strongly that these keys have to be non-updatable, especially for objects that are managed exclusively by admins. I'm fine with any of solutions 1, 3, or 4.

Reply:
Oh good catch, it's not in the design so I missed it. I'll keep things consistent and keep it the same as the other models.

Reply:
Okay this is fixed in [bc5a97d](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12483/commits/bc5a97d97b99e83cb36a4724b8958711878d0b51)